src/CloudLfs.Core/Services/TransferOrchestrationService.cs

```diff
@@ -7,55 +7,105 @@
 
 namespace Microsoft.MixedReality.CloudLfs.Services
 {
-    public class TransferOrchestrationService
+    public class TransferOrchestrationService : ITransferOrchestrationService
     {
         private readonly IGitLfsMessageService _messageService;
 
         private readonly IBlobBroker _blobBroker;
 
         private readonly IGitLfsBroker _lfsBroker;
+        private readonly string _tempPath;
 
-        public TransferOrchestrationService(IGitLfsMessageService messageService, IBlobBroker blobBroker, IGitLfsBroker lfsBroker)
+        public TransferOrchestrationService(IGitLfsMessageService messageService, IBlobBroker blobBroker, IGitLfsBroker lfsBroker, string tempPath = default)
         {
             _messageService = messageService;
             _blobBroker = blobBroker;
             _lfsBroker = lfsBroker;
+            _tempPath = tempPath ?? Path.GetTempPath();
         }
 
         public async Task RunAsync(CancellationToken cancellationToken)
         {
             while (!cancellationToken.IsCancellationRequested)
             {
+                await Task.Delay(10);
+
                 var message = _messageService.ReadMessage();
                 if (message is TerminateTransferGitLfsMessage)
                 {
-                    break;
+                    return;
                 }
                 else if (message is InitializeTransferGitLfsMessage)
                 {
                     _messageService.WriteMessage(new AcknowledgeGitLfsMessage());
                 }
+                else if (message is UploadObjectLfsGitMessage uploadMessage)
```

Comment:
we can extract these into their own functions

Reply:
Will address in next PR.